pkg/pdatagen/pmetricsgen/data.go

```diff
@@ -133,40 +146,64 @@ func Generate(opts ...GenerationOption) pmetric.Metrics {
 // the point attributes will be named as gauge.attr0, gauge.attr1, etc.
 // the scope attributes will be named as scope.attr0, scope.attr1, etc.
 // the resource attributes will be named as resource.attr0, resource.attr1, etc.
-func GenerateGaugeMetrics(numMetrics, numDataPoints, numPointAttributes, numScopeAttributes, numResourceAttributes int) pmetric.Metrics {
+func GenerateGaugeMetrics(numMetrics, numDataPoints, numPointAttributes, numScopeAttributes, numResourceAttributes int, numNanValues int, numNoRecordedValues int) pmetric.Metrics {
 	metrics := pmetric.NewMetrics()
-	rm := metrics.ResourceMetrics().AppendEmpty()
-	for i := 0; i < numResourceAttributes; i++ {
-		rm.Resource().Attributes().PutStr("resource.attr_"+strconv.Itoa(i), "value"+strconv.Itoa(i))
-	}
-	rm.SetSchemaUrl("resource.schema_url")
-	sm := rm.ScopeMetrics().AppendEmpty()
-	sm.SetSchemaUrl("scope.schema_url")
-	for i := 0; i < numScopeAttributes; i++ {
-		sm.Scope().Attributes().PutStr("scope.attr_"+strconv.Itoa(i), "value"+strconv.Itoa(i))
-	}
-	sm.Scope().SetName("go.signoz.io/app/reader")
-	sm.Scope().SetVersion("1.0.0")
+	_, sm := setupResourceAndScope(&metrics, numResourceAttributes, numScopeAttributes, "value", "value")
+
+	const baseTimestamp = 1727286182
+
 	for i := 0; i < numMetrics; i++ {
 		m := sm.Metrics().AppendEmpty()
 		m.SetName("system.memory.usage" + strconv.Itoa(i))
 		m.SetUnit("bytes")
 		m.SetDescription("memory usage of the host")
+
 		dpSlice := m.SetEmptyGauge().DataPoints()
-		for j := 0; j < numDataPoints; j++ {
+		validDataPoints := numDataPoints - numNanValues - numNoRecordedValues
```

Comment:
This would be surprising, without knowing the impl detail one would imagine they get back metrics with `numDataPoints` valid point, `numNanValues`, nan points, and `numNoRecordedValues` no rec points but this. Let's either rename the `numDataPoints` to `numValidDataPoints` or make the func return `numDataPoints + numNanValues + numNoRecordedValues` points